engine/src/main/java/org/terasology/config/ServerInfo.java

```diff
@@ -123,4 +130,19 @@ public String toString() {
         return "ServerInfo [name=" + name + ", address=" + address + ", port=" + port +
                 ", owner=" + owner + ", active=" + active + "]";
     }
+
+    /**
+     *
+     * @param protocol the protocol the server uses
+     * @return the URL of the server
+     */
+    public URL getURL(String protocol) {
```

Comment:
Returning a class instance _or_ a `null`? This calls for an [`Optional`](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html)!